### PR TITLE
Restore adaptive nature to new Scrollbar

### PR DESCRIPTION
**Pull request by Piinks**
This PR makes the `Scrollbar` widget adaptive again.
A large refactoring was completed in https://github.com/flutter/flutter/pull/71664, which temporarily disabled the adaptive behavior of the `Scrollbar` widget. This brings that ability back, in a cleaner way by making the `Scrollbar` widget a wrapper class. The new Material Design scroll bar is in _MaterialScrollbar, keeping it separate from the CupertinoScrollbar.

The adaptability tests that were removed in #71664 have been restored. Added a todo to update the `CupertinoScrollbar` for desktop behavior.

Part of #70866

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test exempt.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I signed the [CLA].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: packages/flutter/lib/src/material/scrollbar.dart [hunk 1]
```diff
@@ -58,6 +61,114 @@ class Scrollbar extends RawScrollbar {
...
+
+class _ScrollbarState extends State<Scrollbar> {
+  late bool _useCupertinoScrollbar;
+
+  @override
+  void didChangeDependencies() {
+    super.didChangeDependencies();
+    final ThemeData theme = Theme.of(context);
+    switch (theme.platform) {
+
```
Comment (HansMuller) [inline packages/flutter/lib/src/material/scrollbar.dart:126]: Extra blank line?

File: packages/flutter/lib/src/material/scrollbar.dart [hunk 2]
```diff
@@ -58,6 +61,114 @@ class Scrollbar extends RawScrollbar {
...
+        break;
+      case TargetPlatform.macOS:
+        // The current CupertinoScrollbar is not intended for desktop.
+        // TODO(Piinks): Update CupertinoScrollbar appearance for desktop.
+        _useCupertinoScrollbar = false;
+        break;
+    }
+  }
+
+  @override	  @override
```
Comment (HansMuller) [inline packages/flutter/lib/src/material/scrollbar.dart:144]: Typo

File: packages/flutter/test/material/scrollbar_test.dart [hunk 3]
```diff
@@ -989,4 +991,72 @@ void main() {
...
+    await tester.pumpWidget(viewWithScroll(debugDefaultTargetPlatformOverride));
+    final TestGesture gesture = await tester.startGesture(
+      tester.getCenter(find.byType(SingleChildScrollView))
+    );
+    await gesture.moveBy(const Offset(0.0, -10.0));
+    await tester.drag(find.byType(SingleChildScrollView), const Offset(0.0, -10.0));
+    await tester.pump();
+    await tester.pump(const Duration(milliseconds: 200));
+    expect(find.byType(CupertinoScrollbar), paints..rrect());
+    final CupertinoScrollbar scrollbar = find.byType(CupertinoScrollbar).evaluate().first.widget as CupertinoScrollbar;
```
Comment (HansMuller) [inline packages/flutter/test/material/scrollbar_test.dart:1059]: Slightly simpler and there should only be one CupertinoScrollbar (right?)
```dart
final CupertinoScrollbar scrollbar = tester.widget<CupertinoScrollbar>(find.byType(CupertinoScrollbar));
```

File: packages/flutter/lib/src/material/scrollbar.dart [hunk 4]
```diff
@@ -58,6 +61,114 @@ class Scrollbar extends RawScrollbar {
...
+        isAlwaysShown: widget.isAlwaysShown ?? false,
+        thickness: widget.thickness ?? CupertinoScrollbar.defaultThickness,
+        thicknessWhileDragging: widget.thickness ?? CupertinoScrollbar.defaultThicknessWhileDragging,
+        radius: widget.radius ?? CupertinoScrollbar.defaultRadius,
+        radiusWhileDragging: widget.radius ?? CupertinoScrollbar.defaultRadiusWhileDragging,
+        controller: widget.controller,
+      );
+    }
+    return _MaterialScrollbar(
+      key: widget.key,
```
Comment (HansMuller) [inline packages/flutter/lib/src/material/scrollbar.dart:158]: Shouldn't be passing along the key here.

File: packages/flutter/lib/src/material/scrollbar.dart [hunk 5]
```diff
@@ -58,6 +61,114 @@ class Scrollbar extends RawScrollbar {
...
+  /// [CupertinoScrollbar.defaultRadius] is used. The remaining platforms have a
+  /// default [Radius.circular] of 8.0 pixels.
+  final Radius? radius;
+
+  @override
+  _ScrollbarState createState() => _ScrollbarState();
+}
+
+class _ScrollbarState extends State<Scrollbar> {
+  late bool _useCupertinoScrollbar;
```
Comment (HansMuller) [inline packages/flutter/lib/src/material/scrollbar.dart:119]: There's no need to cache this. Could just call a `_useCupertinoScrollbar` getter at build time.

File: packages/flutter/test/material/scrollbar_test.dart [hunk 6]
```diff
@@ -1056,7 +1056,6 @@ void main() {
     await tester.pump();
     await tester.pump(const Duration(milliseconds: 200));
     expect(find.byType(CupertinoScrollbar), paints..rrect());
-    final CupertinoScrollbar scrollbar = find.byType(CupertinoScrollbar).evaluate().first.widget as CupertinoScrollbar;
-    expect(scrollbar.controller, isNotNull);
+    final CupertinoScrollbar scrollbar = tester.widget<CupertinoScrollbar>(find.byType(CupertinoScrollbar));    expect(scrollbar.controller, isNotNull);
```
Comment (HansMuller) [inline packages/flutter/test/material/scrollbar_test.dart:1059]: It looks like a line break was left out here

**Review by HansMuller — APPROVED** [review]
LGTM

File: packages/flutter/test/material/scrollbar_test.dart (hunk 6 above)
Comment (Piinks) [reply packages/flutter/test/material/scrollbar_test.dart]: Ah my IDE is unforgivable today. :P

**Review by AMH8790 — COMMENTED** [review]
Mahmas501gmail.com

**Pull request merged**